astropy/table/serialize.py

```diff
@@ -0,0 +1,165 @@
+
+    Normally contains the single key ``name`` with the name of the column in the
+    table.
+    """
+    pass
+
+
+def _represent_mixins_as_columns(tbl):
+    """
+    Convert any mixin columns to plain Column or MaskedColumn and
+    return a new table.
+    """
+    if not tbl.has_mixin_columns:
+        return tbl
+
+    mixin_cols = {}
+
+    new_cols = []
+
+    # Subtlety here is handling mixin info attributes.  The basic list of such
+    # attributes is: 'name', 'unit', 'dtype', 'format', 'description', 'meta'.
+    # - name: handled at the top level [DON'T store]
+    # - unit: DON'T store if this is a parent attribute
+    # - dtype: captured in plain Column if relevant [DON'T store]
+    # - format: possibly irrelevant but settable post-object creation [DO store]
+    # - description: DO store
+    # - meta: DO store
+
+    for col in tbl.itercols():
+        if not has_info_class(col, MixinInfo):
+            new_cols.append(col)
+        else:
+            obj_attrs = col.info._represent_as_dict()
+            data_attrs = col.info._represent_as_dict_data_attrs
+
+            # Set the __info__ attributes (see long note above)
+            info = {}
+            for attr, nontrivial, xform in (('unit', lambda x: x not in (None, ''), str),
+                                            ('format', lambda x: x is not None, None),
+                                            ('description', lambda x: x is not None, None),
+                                            ('meta', lambda x: x, None)):
+                col_attr = getattr(col.info, attr)
+                if nontrivial(col_attr):
+                    info[attr] = xform(col_attr) if xform else col_attr
+
+            for data_attr in data_attrs:
+                data = obj_attrs[data_attr]
+
+                name = col.info.name
+                # New column name is the same as original if there is only one
+                # data attribute (e.g. time.value), otherwise make a new name
+                # (e.g. skycoord.ra, skycoord.dec).
+                #
+                # In the single-name case the mixin Info gets put into the new
+                # Column so the ECSV header looks complete (and for back compatibility
+                # with pre-2.0 behavior).  However, these are ignored in object
+                # construction later in favor of values in the
+                # __mixin_columns__['__info__'] dict.  This is because in general
+                # object constructors don't pay attention to the .info attributes
+                # in the input data args (e.g. Quantity(col) does look at the unit
+                # but not info.description).
+                if len(data_attrs) > 1:
+                    name = name + '.' + data_attr
+                    col_info = {}
+                else:
+                    col_info = info
+                new_cols.append(Column(data, name=name, **col_info))  # TODO masking, MaskedColumn
```

Comment:
Ah, so much nicer than what I had!! So, I would suggest setting `info = {}` for the single-col case.